nexus/db-queries/src/db/datastore/scim_provider_store.rs

```diff
@@ -1386,14 +1464,7 @@ impl<'a> ProviderStore for CrdbScimProviderStore<'a> {
         let err: OptionalError<ProviderStoreError> = OptionalError::new();
 
         let users = self
-            .datastore
-            .transaction_retry_wrapper("scim_list_users")
-            .transaction(&conn, |conn| {
-                let err = err.clone();
-                let filter = filter.clone();
-
-                async move { self.list_users_in_txn(&conn, err, filter).await }
-            })
+            .list_users_with_groups(&conn, err.clone(), filter)
```

Comment:
hmm I think doing this in a txn is load bearing, is that right  @jmpesp ?

Reply:
I thought it was only because there were 50 queries in there. It can easily be put back.